actioncable/lib/action_cable/connection/web_socket.rb

```diff
@@ -17,23 +17,23 @@ def possible?
       end
 
       def alive?
-        websocket && websocket.alive?
+        websocket&.alive?
       end
 
       def transmit(data)
-        websocket.transmit data
+        websocket&.transmit data
       end
 
-      def close
-        websocket.close
+      def close(code = nil, reason = nil)
```

Comment:
This is an internal method, we do not expose it anywhere.

Thus, I'd suggest to hack around `Connection#close` in your codebase instead.

Reply:
With this change, you still need to access internals, e.g., `self.websocket.close(code, reason)`, but you can already do this via `self.websocket.websocket.close(code, reason)`.

 > Could you give me some hints on how to hack Connection#close to customize the code and reason?
 
Smth like:
 
```ruby
def close(ws_code: nil, ws_reason: nil, **)
  return websocket.close(ws_code, ws_reason) if ws_code
  
  super(**)
end

# and then
connection.close(ws_code: 1001, ws_reason: "whatever")
```

Reply:
I don't mean to change the `Connection#close` interface.

I can override the `Connection#close` method with my change.

```ruby
class MyConnection < ActionCable::Connection::Base
  def close
    code = 1001 # Determine which code should return
    websocket.close(code)
  end
end
```

Reply:
In your code

```ruby
def close(ws_code: nil, ws_reason: nil, **)
  return websocket.close(ws_code, ws_reason) if ws_code
  
  super(**)
end
```
My patch is actually made this line `websocket.close(ws_code, ws_reason)` work.

The real `websocket` object allows passing `code` and `reason` on `close`,
it is wrapped by the `ActionCable::Connection::WebSocket` (the `websocket` instance variable in the `Connection` class), which doesn't expose the ability, I'm adding it.